### DISCUSSION TITLE
How to get favicon to show up on PDF files? Please help!

### DISCUSSION THREAD

**Discussion by thealainpaul** [community/community#128701] [New to GitHub] [answered]
### Select Topic Area

Product Feedback

### Body

I am using github for the sole purpose of being able to show my pdfs with my domain/brand in the url. I would also like to show the favicon for the pdfs. I have a 404 page setup incase any one inputs a wrong url. my domain is mywebsite.com hosted on clickfunnels. But clickfunnels doesn't allow for my domain to be displayed in the url for pdfs, so I came to github, where this is possible, and I am using mywebsite.download The very last thing I need to be done with this, is to show the favicon on the pdf files. I have tried putting favicon.ico in the root. This doesn't work for PDFs. Furthermore I am not code savvy, so like a child, I would need help with this. Thanks in advance.

**Comment by TheGiraffe3** [comment]
What type of Operating System do you have?

**Reply by thealainpaul** [reply]
Windows 11 Pro, thanks for responding. Any assistance to make this work will just be fantastic! Best, Alain

**Reply by TheGiraffe3** [reply]
Sorry, I don’t know Windows. What about [this](https://stackoverflow.com/questions/16979659/setting-a-single-favicon-for-the-entire-website) though?

**Reply by thealainpaul** [reply]
I tried the fix at the end, but no go. It doesn't matter if I am using windows or not, because everything can be changed etc with in the files themselves in github, so if you know a method through any other service, it could help... LOL, many years ago I had Kali installed with windows. Kali at the time was deemed a security threat to the powers that be, which in my understanding, meant it was pretty much impenetrable. Anyhow, any assistance here would be not without the most appreciation you can imagine!

**Reply by TheGiraffe3** [reply]
If [this](https://stackoverflow.com/questions/4886423/display-favicon-for-pdf-files#:~:text=If%20you%20put%20a%20favicon,will%20actually%20have%20that%20favicon.) doesn’t work, then I’m not sure. Sorry.

**Reply by thealainpaul** [reply]
Ok no worries! I tried it but doesn't seem to go. Thanks for trying, will also give it a fe wminutes, just in case... Sometimes seems to be a bit of a lag for changes to work anyhow...

**Reply by TheGiraffe3** [reply]
@thealainpaul what luck?

**Reply by thealainpaul** [reply]
@TheGiraffe3 No luck yet mate! I believe, since I saw in n article that some how in code I need to tell the index to put the favicon for pdfs. This is because PDFs aren't webpages, but files, so the code has to support it, but the issue is I can't code, very limited.

**Comment by m1her** [comment]
I'm not sure but try this link instead of the one your using in the link in head tag 

href="https://raw.githubusercontent.com/thealainpaul/thealainpaul.github.io/main/Favicon.ico"

**Reply by thealainpaul** [reply]
> I'm not sure but try this link instead of the one your using in the link in head tag
> 
> href="https://raw.githubusercontent.com/thealainpaul/thealainpaul.github.io/main/Favicon.ico"

The good news is it is using the favicon from my repository now, so it loads faster, the bad news is it still doesn't show on the pdf. Thanks for trying, any other possible ideas?

**Reply by m1her** [reply]
I'm sorry that's all i've got

**Reply by thealainpaul** [reply]
> I'm sorry that's all i've got

No problem! Thanks for givng me your time.

Best, Alain

**Comment by thealainpaul** [comment]
No luck yet, stackoverflow someone explained to put in root, I believe it
already is... But not sure.

**Comment by thealainpaul** [comment]
Anyone else to have a go at this issue? I will truly appreciate it! Thanks in advance!

**Comment by thealainpaul** [comment]
Chatgpt said to create separate html files for each pdf and create redirect links. It professes that normally this works. But I am unable to get it to work. The favicon is in the route and the code is as follows:

```
`<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" type="image/x-icon" href="https://raw.githubusercontent.com/thealainpaul/thealainpaul.github.io/main/Favicon.ico">
    <title>Redirecting to Marketing1.pdf...</title>
    <meta http-equiv="refresh" content="0; url=Marketing1.pdf">
</head>
<body>
    <p>If you are not redirected, <a href="Marketing1.pdf">click here to view Marketing1.pdf</a>.</p>
</body>
</html>`
```

**Comment by thealainpaul** [comment, marked as answer]
I did it with the help of chatgpt through iframe, for each pdf file I created an html file for the pdf. Here is an example of one of the codes:

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" type="image/x-icon" href="https://raw.githubusercontent.com/thealainpaul/thealainpaul.github.io/main/Favicon.ico">
    <title>Marketing 1</title>
    <style>
        body, html {
            margin: 0;
            padding: 0;
            width: 100%;
            height: 100%;
        }
        iframe {
            width: 100%;
            height: 100%;
            border: none;
        }
    </style>
</head>
<body>
    <iframe src="Marketing1.pdf" type="application/pdf"></iframe>
</body>
</html>
```

**Reply by TheGiraffe3** [reply]
Great! I’m glad you finally got it working!

**Comment by EasyCodeCoach** [comment]
Response #20633
